scanner/src/main/kotlin/Scanner.kt

```diff
@@ -141,15 +141,72 @@ class Scanner(
             }
         }
 
+        val filesByProvenance = (projectResults.files + packageResults.files)
+            .groupBy { it.provenance }
+            .values
+            .map { fileLists ->
+                fileLists.reduce { acc, next ->
+                    acc.copy(
+                        files = acc.files + next.files
```

Comment:
Why is it necessary to merge the file lists?
To my understanding, the size of each `groupBy()` resulting group is maximum 2,
and if it is 2, then boths file lists are identical.

..So, could we simply replace lines 144 -151 with the following?
```
(projectResults.files + packageResults.files).distinctBy { it.provenance }.toSet()
```